src/uwp/Microsoft.Build.Net.CoreRuntimeTask/ResourceHandlingTask.cs

```diff
@@ -267,18 +323,82 @@ private void WriteStateFile(string stateFile, ResourceHandlingState state)
+            }
+            else {
+                string culturePath = "";
+                string culture = assembly.Culture;
+                string assemblyName = assembly.Name.Value;
+
+                if (!String.IsNullOrEmpty(culture))
+                {
+                    culturePath = culture + Path.DirectorySeparatorChar;
+                }
+                else if (TryGetNeutralResourcesLanguageAttribute(assembly, out neutralResourceLanguage))
+                {
+                    culturePath = neutralResourceLanguage + Path.DirectorySeparatorChar;
+                }
+                else
+                {
+                    // Do not handle the case where culture is Invariant and no NeutralResourcesLanguageAttribute is declared
+                    // This should already be taken care of in method ExtractAssemblyResWList
+                }
 
-                string reswFilePath = OutResWPath + Path.AltDirectorySeparatorChar + "FxResources." + assemblyName + ".SR.resw";
-                WriteResW(stream, reswFilePath);
-                return reswFilePath;
+                if (resourceFileName.EndsWith("." + culture, StringComparison.OrdinalIgnoreCase))
+                {
+                    resourceFileName = resourceFileName.Remove(resourceFileName.Length - (culture.Length + 1));
+                }
+
+                resourceIndexName = assemblyName.EndsWith(".resources", StringComparison.OrdinalIgnoreCase) ? assemblyName.Remove(assemblyName.Length - 10) : assemblyName;
+                reswFilePath = OutResWPath + resourceIndexName + Path.DirectorySeparatorChar + culturePath + resourceFileName + ".resw";
+                if (!Directory.Exists(Directory.GetParent(reswFilePath).ToString()))
+                {
+                    Directory.CreateDirectory(Directory.GetParent(reswFilePath).ToString());
+                }
             }
+
+            WriteResW(stream, reswFilePath);
+
+            FileInfo fiResW = new FileInfo(reswFilePath);
+            return new ResWInfo() { ResWPath = reswFilePath, ResWTimeUtc = fiResW.LastWriteTimeUtc, ResourceIndexName = resourceIndexName, NeutralResourceLanguage = neutralResourceLanguage };
+        }
+
+        private bool TryGetNeutralResourcesLanguageAttribute(IAssembly assembly, out String neutralResourceLanguage)
+        {
+            neutralResourceLanguage = "";
+            foreach (ICustomAttribute attribute in assembly.AssemblyAttributes)
+            {
+                if (TypeHelper.GetTypeName(attribute.Type, NameFormattingOptions.None).Equals("System.Resources.NeutralResourcesLanguageAttribute"))
+                {
+                    if (attribute.Arguments.Count() > 0)
+                    {
+                        IMetadataConstant metadataConstant = attribute.Arguments.ElementAt(0) as IMetadataConstant;
+                        if (metadataConstant == null)
+                        {
+                            return false; // Unable to parse
```

Comment:
> return false; // Unable to parse [](start = 27, length = 33)

should we continue the loop instead of returning false? just in case we have multiple neutral attribute definition? I am not sure if we can have such case though.

Reply:
The VS UI will only set one attribute, so unless people hand-write the NeutralResourcesAttribute then I think we'll be fine. I think it makes sense to return.